packages/fx-core/src/plugins/resource/cicd/index.ts

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import {
+  err,
+  UserError,
+  SystemError,
+  v2,
+  TokenProvider,
+  FxError,
+  Result,
+  Inputs,
+  Json,
+  Func,
+  ok,
+  QTreeNode,
+  ProjectSettings,
+  SingleSelectQuestion,
+  Platform,
+} from "@microsoft/teamsfx-api";
+
+import { FxResult, FxCICDPluginResultFactory as ResultFactory } from "./result";
+import { CICDImpl } from "./plugin";
+import { ErrorType, InternalError, NoProjectOpenedError, PluginError } from "./errors";
+import { LifecycleFuncNames } from "./constants";
+import { Service } from "typedi";
+import { ResourcePluginsV2 } from "../../solution/fx-solution/ResourcePluginContainer";
+import { ResourcePlugin, Context, DeepReadonly } from "@microsoft/teamsfx-api/build/v2";
+import {
+  githubOption,
+  azdoOption,
+  jenkinsOption,
+  ciOption,
+  cdOption,
+  provisionOption,
+  publishOption,
+  questionNames,
+} from "./questions";
+import { Logger } from "./logger";
+import { environmentManager } from "../../..";
```

Comment:
To avoid circular reference, we don't recommend import from folder.

Reply:
i found this useful when i had to debug circularity once https://www.npmjs.com/package/madge

Reply:
It's auto generated, I believe. Let me refine it soon.